tests/test_freeze.py

```diff
@@ -70,12 +72,13 @@ def func(x, y):
 
 
 @pytest.test_arrays("uint32, jit, shape=(*)")
-def test03_output_tuple(t):
+@pytest.mark.parametrize("auto_opaque", [False, True])
+def test03_output_tuple(t, auto_opaque):
     """
     Tests that returning tuples from frozen functions is possible.
     """
 
-    @dr.freeze
+    @dr.freeze(auto_opaque = auto_opaque)
```

Comment:
```suggestion
    @dr.freeze(auto_opaque=auto_opaque)
```

and same everywhere below